src/components/numeral-date/numeral-date.pw.tsx

```diff
@@ -0,0 +1,685 @@
+    await expect(tooltipPreview(page)).toHaveAttribute(
+      "data-placement",
+      position as string
+    );
+  });
+});
+
+test("should render NumeralDate with helpAriaLabel prop", async ({
+  mount,
+  page,
+}) => {
+  await mount(
+    <NumeralDateComponent
+      labelHelp="fieldHelp"
+      helpAriaLabel={CHARACTERS.STANDARD}
+    />
+  );
+
+  const help = getDataComponentByValue(page, "help");
+
+  await expect(help).toHaveAttribute("aria-label", CHARACTERS.STANDARD);
+});
+
+test("should have the expected border radius styling when no search button enabled", async ({
+  mount,
+  page,
+}) => {
+  await mount(<NumeralDateComponent />);
+  await expect(numeralDateInput(page, 0).locator("..")).toHaveCSS(
+    "border-radius",
+    "4px 0px 0px 4px"
+  );
+  await expect(numeralDateInput(page, 2).locator("..")).toHaveCSS(
+    "border-radius",
+    "0px 4px 4px 0px"
+  );
+});
+
+test.describe("check events for NumeralDate component", () => {
+  test("should call onChange callback when a type event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    const inputValue = "1";
+    let callbackCount = 0;
+    await mount(
+      <NumeralDateComponent
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await numeralDateInput(page, 0).type(inputValue);
+
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onBlur callback when a blur event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <NumeralDateComponent
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const input = numeralDateInput(page, 0);
+    await input.focus();
+    await input.blur();
+
+    await page.waitForTimeout(50);
+
+    await expect(input).not.toBeFocused();
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("check accessibility for NumeralDate component", () => {
+  test("should pass accessibility tests for NumeralDate component", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<NumeralDateComponent />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component ", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<NumeralDateControlled />);
+
+    await checkAccessibility(page);
+  });
+
+  ([
+    ["dd", "mm", "yyyy"],
+    ["mm", "dd", "yyyy"],
+    ["dd", "mm"],
+    ["mm", "dd"],
+    ["mm", "yyyy"],
+  ] as NumeralDateProps["dateFormat"][]).forEach((dateFormat) => {
+    test(`should pass accessibility tests for NumeralDate when dateFormat prop is ${dateFormat}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<NumeralDateComponent dateFormat={dateFormat} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with enableInternalError prop", async ({
```

Comment:
comment: You probably want to trigger the validation tooltip here and then run the check